### DISCUSSION TITLE
Location of mysql.sock

### DISCUSSION THREAD

**Discussion by ernstpijper** [DIRACGrid/DIRAC#6259] [Q&A] [answered]
Hi,

DIRAC py3 server 7.3.17, py3 pilot jobs and py3 diracos2 client

There was mention of this problem before, here

DIRAC py3 server 7.3.17, py3 pilot jobs and py3 diracos2 client

I ran into this problem as well when installng DIRAC. The default location of the mysql socket is /var/lib/mysql/mysql.sock. But the DIRAC processes want to use /tmp/mysql.sock. What i did is to create a link: 

/tmp/mysql.sock -> /var/lib/mysql/mysql.sock

Last week, this link, for yet unknown reasons, disappeared. This resulted in the following error messages:

2022-07-06 19:39:01 UTC Framework/ProxyManager NOTICE: Returning response ([::ffff:145.38.219.85]:38532)[hosts:nl-dirac-wms] (275.27 secs) ERROR: MySQL Error ( 1131 : Could not connect: (2002, "Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)"))

After I recreated the link, it worked again.

Is there a way to configure the socket location for the dirac processes?

**Comment by IgorPelevanyuk** [comment, marked as answer]
Hi, I also faced with this problem on DIRAC 8.0.0a26 and MariaDB 10.5.16. Reconfiguring of /etc/my.cnf helped.
```
[client]
port = 3306
socket = /tmp/mysql.sock

[mysqld]
port = 3306
socket = /tmp/mysql.sock
```
